sklearn/metrics/_classification.py

```diff
@@ -2360,6 +2360,18 @@ def recall_score(
     ``UndefinedMetricWarning``. This behavior can be modified with
     ``zero_division``.
 
+    When used with ``average=macro``, the result equals that of
+    :func:`balanced_accuracy_score`. That is,
+    ``recall_score(y_true, y_pred, average=macro)`` is equivalent to
+    ``balanced_accuracy_score(y_true, y_pred)``.
+
+    References
+    ----------
+    .. [1] J. Opitz (2024). `"A Closer Look at Classification Evaluation Metrics
+           and a Critical Reflection of Common Evaluation Practice". Transactions
+           of the Association for Computational Linguistics 12 (2024): 820-836
+           <https://doi.org/10.1162/tacl_a_00675>`_.
```

Comment:
So, at the end of the day, it comes down to tracing the definitions:

$$\mbox{Macro-averaged Recall} := \displaystyle \sum_{i=1}^{n_{\mathrm{classes}}} \mbox{Recall for class $i$} =: \mbox{Balanced Accuracy}$$

However, it's not clear to me how the newly added reference justify this, or at least, provide a rigourious proof on the equivalence of two metrics.

Maybe pinging @glemaitre for a second opinion here.

Reply:
Thanks @virchan for checking into this. I think the proof that you were looking for is in Eq. 8 in the referenced paper.

Sorry, I guess I maybe should've highlighted this in the opening post.